src/ui/public/state_management/state_storage/_tests__/hashed_item_store.js

```diff
@@ -0,0 +1,334 @@
+import expect from 'expect.js';
+import sinon from 'sinon';
+import bluebird from 'bluebird';
+
+import StubBrowserStorage from 'test_utils/stub_browser_storage';
+import HashedItemStore from '../hashed_item_store';
+
+describe('hashedItemStore', () => {
+  describe('interface', () => {
+    describe('#constructor', () => {
+      it('retrieves persisted index from sessionStorage', () => {
+        const sessionStorage = new StubBrowserStorage();
+        sinon.spy(sessionStorage, 'getItem');
+
+        const hashedItemStore = new HashedItemStore(sessionStorage);
```

Comment:
I wouldn't assign the reference if we're not using it. We're testing for a side-effect anyways.